### ISSUE TITLE
ORCA signal 9 SIGSEGV (null-pointer dereference)

### ISSUE THREAD

**Issue by dh-cloud** [greenplum-db/gpdb#11660]
### Greenplum version or build

Greenplum Database 6.11.2

### OS version and uname -a

CentOS 7.7.1908 x64

### Installation information ( pg_config )

```
# select * from gp_segment_configuration ;
 dbid | content | role | preferred_role | mode | status | port  | hostname | address |        datadir        
------+---------+------+----------------+------+--------+-------+----------+---------+-----------------------
    1 |      -1 | p    | p              | n    | u      |  5432 | mdw      | mdw     | /data1/master/gpseg-1
    2 |       0 | p    | p              | n    | u      | 40000 | sdw1     | sdw1    | /data1/primary/gpseg0
    3 |       1 | p    | p              | n    | u      | 40001 | sdw1     | sdw1    | /data2/primary/gpseg1
    4 |       2 | p    | p              | n    | u      | 40002 | sdw1     | sdw1    | /data3/primary/gpseg2
(4 rows)
```

### Expected behavior

No Crash

### Actual behavior

Aim: test vmem runaway, so i set  gp_vmem_protect_limit to **50MB**:

```
gpconfig -c gp_vmem_protect_limit -v 50
gpstop -r -a

create table foo(i int, j int);
```

psql

```
cloud=# insert into foo select generate_series(20,30);
ERROR:  Canceling query because of high VMEM usage. Used: 24MB, available 0MB, red zone: 45MB (runaway_cleaner.c:215)
cloud=# insert into foo select generate_series(20,30);
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.
!> \q

```

### Step to reproduce the behavior

Core dump:

```
Core was generated by `postgres:  5432, gpadmin cloud [local] con9 cmd6 INSERT                       '.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fea7072f4bb in raise () from /lib64/libpthread.so.0
[Current thread is 1 (Thread 0x7fea72412880 (LWP 31815))]
Missing separate debuginfos, use: debuginfo-install bzip2-libs-1.0.6-13.el7.x86_64 cyrus-sasl-lib-2.1.26-23.el7.x86_64 glibc-2.17-292.el7.x86_64 keyutils-libs-1.5.8-3.el7.x86_64 krb5-libs-1.15.1-37.el7_6.x86_64 libcom_err-1.42.9-16.el7.x86_64 libcurl-7.29.0-54.el7.x86_64 libgcc-4.8.5-39.el7.x86_64 libidn-1.28-4.el7.x86_64 libselinux-2.5-14.1.el7.x86_64 libssh2-1.8.0-3.el7.x86_64 libxml2-2.9.1-6.el7_2.3.x86_64 nspr-4.21.0-1.el7.x86_64 nss-3.44.0-4.el7.x86_64 nss-softokn-freebl-3.44.0-5.el7.x86_64 nss-util-3.44.0-3.el7.x86_64 openldap-2.4.44-21.el7_6.x86_64 pcre-8.32-17.el7.x86_64 xz-libs-5.2.2-1.el7.x86_64 zlib-1.2.7-18.el7.x86_64
(gdb) bt
#0  0x00007fea7072f4bb in raise () from /lib64/libpthread.so.0
#1  0x0000000000e05734 in StandardHandlerForSigillSigsegvSigbus_OnMainThread (processName=0x13cb37b "Master process", postgres_signal_arg=11) at elog.c:5548
#2  0x0000000000c90932 in CdbProgramErrorHandler (postgres_signal_arg=11) at postgres.c:3551
#3  <signal handler called>
#4  0x0000000000f8d19d in gpos::CHashMap<unsigned short const, gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*), &gpdxl::GetHashXMLStr, &gpdxl::IsXMLStrEqual, &(void gpos::CleanupNULL<unsigned short const>(unsigned short const*)), &(void gpos::CleanupNULL<gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)>(gpdxl::CParseHandlerBase* (*)(gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)))>::GetChain(unsigned short const*) const (this=0x0, key=0x4365fa8) at ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:150
#5  0x0000000000f8d670 in gpos::CHashMap<unsigned short const, gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*), &gpdxl::GetHashXMLStr, &gpdxl::IsXMLStrEqual, &(void gpos::CleanupNULL<unsigned short const>(unsigned short const*)), &(void gpos::CleanupNULL<gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)>(gpdxl::CParseHandlerBase* (*)(gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)))>::Lookup(unsigned short const*) const (this=0x0, key=0x4365fa8) at ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:172
#6  0x0000000000f8d14d in gpos::CHashMap<unsigned short const, gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*), &gpdxl::GetHashXMLStr, &gpdxl::IsXMLStrEqual, &(void gpos::CleanupNULL<unsigned short const>(unsigned short const*)), &(void gpos::CleanupNULL<gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)>(gpdxl::CParseHandlerBase* (*)(gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)))>::Find(unsigned short const*) const (this=0x0, key=0x4365fa8) at ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:240
#7  0x0000000000f86f19 in gpdxl::CParseHandlerFactory::GetParseHandler (mp=0x35c8fe0, token_identifier_str=0x4365fa8, parse_handler_mgr=0x435b9d0, parse_handler_root=0x435bab8) at CParseHandlerFactory.cpp:314
#8  0x0000000000f8377e in gpdxl::CParseHandlerDXL::StartElement (this=0x435bab8, element_uri=0x4365598, element_local_name=0x4365fa8, element_qname=0x4365f18, attrs=...) at CParseHandlerDXL.cpp:396
#9  0x0000000000f7dd26 in gpdxl::CParseHandlerBase::startElement (this=0x435bab8, element_uri=0x4365598, element_local_name=0x4365fa8, element_qname=0x4365f18, attrs=...) at CParseHandlerBase.cpp:118
#10 0x00007fea71e7142e in xercesc_3_1::SAX2XMLReaderImpl::startElement (this=0x434b000, elemDecl=..., elemURLId=6, elemPrefix=0x43517b8, attrList=..., attrCount=<optimized out>, isEmpty=false, isRoot=false) at ../../src/xercesc/parsers/SAX2XMLReaderImpl.cpp:764
#11 0x00007fea71e1f154 in xercesc_3_1::IGXMLScanner::scanStartTagNS (this=0x434c788, gotData=@0x7ffd8ebdc3d7: true) at ../../src/xercesc/framework/XMLBuffer.hpp:171
#12 0x00007fea71e20ce5 in xercesc_3_1::IGXMLScanner::scanContent (this=0x434c788) at ../../src/xercesc/internal/IGXMLScanner.cpp:890
#13 0x00007fea71e20eb0 in xercesc_3_1::IGXMLScanner::scanDocument (this=0x434c788, src=...) at ../../src/xercesc/internal/IGXMLScanner.cpp:217
#14 0x00007fea71e71c59 in xercesc_3_1::SAX2XMLReaderImpl::parse (this=0x434b000, source=...) at ../../src/xercesc/parsers/SAX2XMLReaderImpl.cpp:409
#15 0x0000000000ff658b in gpdxl::CDXLUtils::GetParseHandlerForDXLString (mp=0x35c8fe0, 
    dxl_string=0x434a760 "<?xml version=\"1.0\" encoding=\"UTF-8\"?><dxl:DXLMessage xmlns:dxl=\"http://greenplum.com/dxl/2010/12/\"><dxl:Metadata><dxl:GPDBFunc Mdid=\"0.1067.1.0\" Name=\"generate_series\" ReturnsSet=\"true\" Stability=\"Im"..., xsd_file_path=0x0)
    at CDXLUtils.cpp:117
#16 0x0000000000ff6cfa in gpdxl::CDXLUtils::GetParseHandlerForDXLString (mp=0x35c8fe0, dxl_string=0x4329e40, xsd_file_path=0x0) at CDXLUtils.cpp:267
#17 0x0000000000ff7d10 in gpdxl::CDXLUtils::ParseDXLToIMDIdCacheObj (mp=0x35c8fe0, dxl_string=0x4329e40, xsd_file_path=0x0) at CDXLUtils.cpp:773
#18 0x0000000001055dec in gpopt::CMDAccessor::GetImdObj (this=0x7ffd8ebdcfd0, mdid=0x43295b8) at CMDAccessor.cpp:537
#19 0x00000000010567b5 in gpopt::CMDAccessor::RetrieveFunc (this=0x7ffd8ebdcfd0, mdid=0x43295b8) at CMDAccessor.cpp:756
#20 0x00000000011ad9c5 in gpdxl::CTranslatorUtils::IsSirvFunc (mp=0x35c8f88, md_accessor=0x7ffd8ebdcfd0, func_oid=1067) at CTranslatorUtils.cpp:216
#21 0x00000000011bca00 in gpdxl::CTranslatorQueryToDXL::HasSirvFunctions (this=0x7ffd8ebdca60, node=0x433ea40) at CTranslatorQueryToDXL.cpp:349
#22 0x00000000011bc959 in gpdxl::CTranslatorQueryToDXL::CheckSirvFuncsWithoutFromClause (this=0x7ffd8ebdca60, query=0x433e8f0) at CTranslatorQueryToDXL.cpp:321
#23 0x00000000011bc4bf in gpdxl::CTranslatorQueryToDXL::CTranslatorQueryToDXL (this=0x7ffd8ebdca60, context=0x431fb78, md_accessor=0x7ffd8ebdcfd0, var_colid_mapping=0x4323978, query=0x433e8f0, query_level=1, is_top_query_dml=true, query_level_to_cte_map=0x431fed0)
    at CTranslatorQueryToDXL.cpp:189
#24 0x00000000011c70c3 in gpdxl::CTranslatorQueryToDXL::TranslateDerivedTablesToDXL (this=0x4323e90, rte=0x433e7d8, rt_index=2, current_query_level=0) at CTranslatorQueryToDXL.cpp:3693
#25 0x00000000011c515d in gpdxl::CTranslatorQueryToDXL::TranslateFromClauseToDXL (this=0x4323e90, node=0x433eeb8) at CTranslatorQueryToDXL.cpp:3044
#26 0x00000000011c4ce7 in gpdxl::CTranslatorQueryToDXL::TranslateFromExprToDXL (this=0x4323e90, from_expr=0x433ee20) at CTranslatorQueryToDXL.cpp:2925
#27 0x00000000011bd199 in gpdxl::CTranslatorQueryToDXL::TranslateSelectProjectJoinToDXL (this=0x4323e90, target_list=0x433eed8, from_expr=0x433ee20, sort_group_attno_to_colid_mapping=0x4327ab8, output_attno_to_colid_mapping=0x43284f8, group_clause=0x0)
    at CTranslatorQueryToDXL.cpp:611
#28 0x00000000011c23ae in gpdxl::CTranslatorQueryToDXL::TranslateGroupingSets (this=0x4323e90, from_expr=0x433ee20, target_list=0x433eed8, group_clause=0x0, has_aggs=false, sort_grpref_to_colid_mapping=0x4327ab8, output_attno_to_colid_mapping=0x4327fd8)
    at CTranslatorQueryToDXL.cpp:2219
#29 0x00000000011bcf8e in gpdxl::CTranslatorQueryToDXL::TranslateSelectQueryToDXL (this=0x4323e90) at CTranslatorQueryToDXL.cpp:557
#30 0x00000000011bd392 in gpdxl::CTranslatorQueryToDXL::TranslateInsertQueryToDXL (this=0x4323e90) at CTranslatorQueryToDXL.cpp:704
#31 0x00000000011bd2dd in gpdxl::CTranslatorQueryToDXL::TranslateQueryToDXL (this=0x4323e90) at CTranslatorQueryToDXL.cpp:670
#32 0x00000000011dcb10 in COptTasks::OptimizeTask (ptr=0x7ffd8ebdd8f0) at COptTasks.cpp:588
#33 0x0000000000f265d4 in gpos::CTask::Execute (this=0x41e7ee8) at CTask.cpp:129
#34 0x0000000000f27fd9 in gpos::CWorker::Execute (this=0x7ffd8ebdd720, task=0x41e7ee8) at CWorker.cpp:80
#35 0x0000000000f25590 in gpos::CAutoTaskProxy::Execute (this=0x7ffd8ebdd750, task=0x41e7ee8) at CAutoTaskProxy.cpp:285
#36 0x0000000000f29712 in gpos_exec (params=0x7ffd8ebdd830) at _api.cpp:246
#37 0x00000000011dbc20 in COptTasks::Execute (func=0x11dc834 <COptTasks::OptimizeTask(void*)>, func_arg=0x7ffd8ebdd8f0) at COptTasks.cpp:257
#38 0x00000000011dd7cf in COptTasks::GPOPTOptimizedPlan (query=0x367d9b8, gpopt_context=0x7ffd8ebdd8f0) at COptTasks.cpp:803
#39 0x00000000011de543 in CGPOptimizer::GPOPTOptimizedPlan (query=0x367d9b8, had_unexpected_failure=0x7ffd8ebdd9af) at CGPOptimizer.cpp:56
#40 0x00000000011deaf3 in GPOPTOptimizedPlan (query=0x367d9b8, had_unexpected_failure=0x7ffd8ebdd9af) at CGPOptimizer.cpp:212
--Type <RET> for more, q to quit, c to continue without paging--
#41 0x0000000000bbc50e in optimize_query (parse=0x35a4490, boundParams=0x0) at orca.c:138
#42 0x0000000000b9f6d8 in standard_planner (parse=0x35a4490, cursorOptions=0, boundParams=0x0) at planner.c:238
#43 0x0000000000b9f649 in planner (parse=0x35a4490, cursorOptions=0, boundParams=0x0) at planner.c:200
#44 0x0000000000c8c4e9 in pg_plan_query (querytree=0x35a4490, cursorOptions=0, boundParams=0x0) at postgres.c:959
#45 0x0000000000c8c5a9 in pg_plan_queries (querytrees=0x367c0d8, cursorOptions=0, boundParams=0x0) at postgres.c:1018
#46 0x0000000000c8dbb2 in exec_simple_query (query_string=0x35a33c8 "insert into foo select generate_series(20,30);") at postgres.c:1752
#47 0x0000000000c92c7a in PostgresMain (argc=1, argv=0x3582e08, dbname=0x3582c70 "cloud", username=0x3582c50 "gpadmin") at postgres.c:5253
#48 0x0000000000bfe7c8 in BackendRun (port=0x35aa290) at postmaster.c:4811
#49 0x0000000000bfde59 in BackendStartup (port=0x35aa290) at postmaster.c:4468
#50 0x0000000000bf9d50 in ServerLoop () at postmaster.c:1948
#51 0x0000000000bf929c in PostmasterMain (argc=6, argv=0x3580cd0) at postmaster.c:1518
#52 0x0000000000af2163 in main (argc=6, argv=0x3580cd0) at main.c:245
(gdb) f 4
#4  0x0000000000f8d19d in gpos::CHashMap<unsigned short const, gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*), &gpdxl::GetHashXMLStr, &gpdxl::IsXMLStrEqual, &(void gpos::CleanupNULL<unsigned short const>(unsigned short const*)), &(void gpos::CleanupNULL<gpdxl::CParseHandlerBase* (gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)>(gpdxl::CParseHandlerBase* (*)(gpos::CMemoryPool*, gpdxl::CParseHandlerManager*, gpdxl::CParseHandlerBase*)))>::GetChain(unsigned short const*) const (this=0x0, key=0x4365fa8) at ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:150
150			return &m_chains[HashFn(key) % m_num_chains];
(gdb) 
(gdb) f 7
#7  0x0000000000f86f19 in gpdxl::CParseHandlerFactory::GetParseHandler (mp=0x35c8fe0, token_identifier_str=0x4365fa8, parse_handler_mgr=0x435b9d0, parse_handler_root=0x435bab8) at CParseHandlerFactory.cpp:314
314			m_token_parse_handler_func_map->Find(token_identifier_str);
(gdb) p m_token_parse_handler_func_map
$2 = (gpdxl::CParseHandlerFactory::TokenParseHandlerFuncMap *) 0x0
```

GPDB LOG:

```
2021-03-16 16:39:54.988686 CST,"gpadmin","cloud",p31815,th1916872832,"[local]",,2021-03-16 16:34:47 CST,0,con9,cmd5,seg-1,,,,sx1,"ERROR","XX000","Canceling query because of high VMEM usage. Used: 24MB, available 0MB, red zone: 45MB (runaway_cleaner.c:215)",,,,,,"insert into foo select generate_series(20,30);",0,,"runaway_cleaner.c",215,"Stack trace:
1    0xdfb525 postgres errstart (elog.c:557)
2    0xe45878 postgres RunawayCleaner_StartCleanup (runaway_cleaner.c:213)
3    0xe45004 postgres RedZoneHandler_DetectRunawaySession (redzone_handler.c:391)
4    0xe4426b postgres VmemTracker_ReserveVmem (vmem_tracker.c:563)
5    0xe43113 postgres <symbol not found> (memprot.c:445)
6    0xe43227 postgres gp_malloc (memprot.c:479)
7    0xe3946c postgres <symbol not found> (aset.c:1360)
8    0xe39a44 postgres <symbol not found> (aset.c:1452)
9    0xe3d7bc postgres MemoryContextAlloc (mcxt.c:1188)
10   0x11ea104 postgres _ZN4gpdb22GPDBMemoryContextAllocEP17MemoryContextDatam (gpdbwrappers.cpp:2713)
11   0x11dde5d postgres _ZN4gpos17CMemoryPoolPalloc7NewImplEjPKcjNS_11CMemoryPool15EAllocationTypeE (discriminator 4)
12   0xf23ac5 postgres _ZN4gpos11CMemoryPool12NewArrayImplIwEEPT_mPKcj (CMemoryPool.h:126)
13   0xf23881 postgres _ZN4gpos13CWStringConstC2EPNS_11CMemoryPoolEPKw (CWStringConst.cpp:64)
14   0xff59be postgres _ZN5gpdxl10CDXLTokens4InitEPN4gpos11CMemoryPoolE (dxltokens.cpp:775)
15   0xffc637 postgres InitDXL (discriminator 2)
16   0x11dbbc2 postgres _ZN9COptTasks7ExecuteEPFPvS0_ES0_ (COptTasks.cpp:242)
17   0x11dd7cf postgres _ZN9COptTasks18GPOPTOptimizedPlanEP5QueryP11SOptContext (COptTasks.cpp:804)
18   0x11de543 postgres _ZN12CGPOptimizer18GPOPTOptimizedPlanEP5QueryPb (CGPOptimizer.cpp:56)
19   0x11deaf3 postgres GPOPTOptimizedPlan (CGPOptimizer.cpp:213)
20   0xbbc50e postgres optimize_query (orca.c:138)
21   0xb9f6d8 postgres standard_planner (planner.c:238)
22   0xb9f649 postgres planner (planner.c:200)
23   0xc8c4e9 postgres pg_plan_query (postgres.c:959)
24   0xc8c5a9 postgres pg_plan_queries (postgres.c:1018)
25   0xc8dbb2 postgres <symbol not found> (postgres.c:1752)
26   0xc92c7a postgres PostgresMain (postgres.c:5255)
27   0xbfe7c8 postgres <symbol not found> (postmaster.c:5442)
28   0xbfde59 postgres <symbol not found> (postmaster.c:4472)
29   0xbf9d50 postgres <symbol not found> (postmaster.c:1954)
30   0xbf929c postgres PostmasterMain (postmaster.c:1518)
"
2021-03-16 16:39:54.988983 CST,"gpadmin","cloud",p31815,th1916872832,"[local]",,2021-03-16 16:34:47 CST,0,con9,cmd5,seg-1,,,,sx1,"LOG","00000","An exception was encountered during the execution of statement: insert into foo select generate_series(20,30);",,,,,,,0,,,,
2021-03-16 16:39:54.989026 CST,"gpadmin","cloud",p31815,th1916872832,"[local]",,2021-03-16 16:34:47 CST,0,con9,cmd5,seg-1,,,,,"LOG","00000","Logging memory usage because of runaway cleanup. Note, this is a post-cleanup logging and may be incomplete.",,,,,,,0,,,,
!! DELETE: memory and context dump !!

2021-03-16 16:40:12.932027 CST,,,p31815,th0,,,2021-03-16 16:34:47 CST,0,con9,cmd6,seg-1,,,,,"PANIC","XX000","Unexpected internal error: Master process received signal SIGSEGV",,,,,,,0,,,,"1    0xe056ef postgres StandardHandlerForSigillSigsegvSigbus_OnMainThread (elog.c:5539)
2    0xc90932 postgres CdbProgramErrorHandler (postgres.c:3551)
3    0x7fea7072f5f0 libpthread.so.0 <symbol not found> + 0x7072f5f0
4    0xf8d670 postgres _ZNK4gpos8CHashMapIKtFPN5gpdxl17CParseHandlerBaseEPNS_11CMemoryPoolEPNS2_20CParseHandlerManagerES4_EXadL_ZNS2_13GetHashXMLStrEPS1_EEXadL_ZNS2_13IsXMLStrEqualESA_SA_EEXadL_ZNS_11CleanupNULLIS1_EEvPT_EEXadL_ZNSB_IS9_EEvSD_EEE6LookupESA_ (CHashMap.h:172)
5    0xf8d14d postgres _ZNK4gpos8CHashMapIKtFPN5gpdxl17CParseHandlerBaseEPNS_11CMemoryPoolEPNS2_20CParseHandlerManagerES4_EXadL_ZNS2_13GetHashXMLStrEPS1_EEXadL_ZNS2_13IsXMLStrEqualESA_SA_EEXadL_ZNS_11CleanupNULLIS1_EEvPT_EEXadL_ZNSB_IS9_EEvSD_EEE4FindESA_ (CHashMap.h:240)
6    0xf86f19 postgres _ZN5gpdxl20CParseHandlerFactory15GetParseHandlerEPN4gpos11CMemoryPoolEPKtPNS_20CParseHandlerManagerEPNS_17CParseHandlerBaseE (CParseHandlerFactory.cpp:314)
7    0xf8377e postgres _ZN5gpdxl16CParseHandlerDXL12StartElementEPKtS2_S2_RKN11xercesc_3_110AttributesE (CParseHandlerDXL.cpp:396)
8    0xf7dd26 postgres _ZN5gpdxl17CParseHandlerBase12startElementEPKtS2_S2_RKN11xercesc_3_110AttributesE (CParseHandlerBase.cpp:119)
"
```

**Comment by zellerh** [comment]
Thank you @dh-cloud, for this bug report! On my machine I didn't see the same failure, but fortunately you included a lot of details, hopefully enough to find the cause of your failure. So far I think the issue is in the `InitDXL()` function. When this fails, we may crash on a second query. Hope to have a PR with a fix, soon.

**Comment by zellerh** [comment]
See the third commit in PR https://github.com/greenplum-db/gpdb/pull/11806 for the fix on the master branch. PRs for 6X_STABLE and probably 5X_STABLE will come once https://github.com/greenplum-db/gpdb/pull/11806 is approved. @dh-cloud, do you build GPDB from source code? If so, could you try the proposed fix (moving the `m_ulpInitDXL++;` line down in method `InitDXL()`)? If not that's ok, I tested it by simulating a similar failure and got the same stack trace. Then I validated that the fix corrects the simulated problem.

**Comment by dh-cloud** [comment]
@zellerh This patch looks good to me,  i think your test validation is enough.

**Comment by zellerh** [comment]
Thanks, @dh-cloud. Submitted PR #11822 for 6X_STABLE.

**Comment by zellerh** [comment]
https://github.com/greenplum-db/gpdb/pull/11822 was merged into 6X_STABLE and a similar fix https://github.com/greenplum-db/gpdb/pull/11806 went into master.
